dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -286,9 +286,7 @@ private void logRetry(long backoffNanoseconds, @Nullable Throwable throwable) {
 
         private ListenableFuture<Response> wrap(ListenableFuture<Response> input) {
             ListenableFuture<Response> result = input;
-            if (!shouldPropagateQos(serverQoS)) {
```

Comment:
subtle bug where this prevented 500s from being retried when qos propagation was enabled.